app-backend/datamodel/src/main/scala/com/azavea/rf/datamodel/ColorCorrect.scala

```diff
@@ -153,25 +153,65 @@ object ColorCorrect {
     val maybeEqualize =
       if (params.equalize) Some(HistogramEqualization(_: MultibandTile, rgbHist)) else None
 
+    case class LayerClipping(redMin: Int, redMax: Int, greenMin: Int, greenMax:Int, blueMin: Int, blueMax: Int)
+    case class ClippingParams(band: Int, min: Option[Int], max: Option[Int], layerMin: Int, layerMax: Int)
+   
+    val layerRgbClipping = (for {
+      (r :: g :: b :: xs) <- Some(rgbHist.toList)
+      isCorrected   <- Some((r :: g :: b :: Nil).foldLeft(true) {(acc, hst) => (
+                          acc && (hst match {
+                            case x if 1 until 255 contains x.minValue().map(_.toInt).getOrElse(0) => true
+                            case x if 1 until 255 contains x.maxValue().map(_.toInt).getOrElse(0) => true
```

Comment:
Yeah I had the same thought process roughly; my initial comment was actually going to be that the range would need to be 0 - 254 in order for the logic to work properly, but then I realized that `until` isn't inclusive; I think a switch to 255 would make the point clearer since `to` and `until` seem easy to confuse.